index.ts

```diff
@@ -0,0 +1,552 @@
+  recharge?: boolean; // If this move is successful, the user must recharge on the following turn and cannot make a move.
+  reflectable?: boolean; // Bounced back to the original user by Magic Coat or the Ability Magic Bounce.
+  snatch?: boolean; // Can be stolen from the original user and instead used by another Pokemon using Snatch.
+  sound?: boolean; // Has no effect on Pokemon with the Ability Soundproof.
+}
+
+type MoveCategory: 'Physical' | 'Special' | 'Status';
+
+type MoveTarget =
+  // single-target
+  'normal'|'any'|'adjacentAlly'|'adjacentFoe'|'adjacentAllyOrSelf'|
+  // single-target, automatic
+  'self'|'randomNormal'|
+  // spread
+  'allAdjacent'|'allAdjacentFoes'|
+  // side and field
+  'allySide'|'foeSide'|'all';
+
+interface SelfEffect {
+  volatileStatus?: ID;
+  boosts?: Partial<BoostsTable>;
+}
+
+interface SecondaryEffect {
+  chance?: number;
+  boosts?: Partial<BoostsTable>;
+  self?: SelfEffect;
+  status?: StatusName;
+  volatileStatus?: ID;
+}
+
+type ContestType = 'Cool' | 'Beautiful' | 'Cute' | 'Clever' | 'Tough';
+
+interface CommonMove<TypeT, ItemT, MoveT> {
+  accuracy: number | true;
+  basePower: number;
+  flags: MoveFlags;
+  pp: number;
+  target: MoveTarget;
+  type: TypeT;
+
+  alwaysCrit?: boolean;
+  alwaysHit?: boolean;
+  basePowerModifier?: number;
+  boosts?: Partial<BoostsTable> | false
+  bypassesProtect?: boolean;
+  category: MoveCategory;
+  contestType?: ContestType;
+  critModifier?: number;
+  critRatio?: number;
+  damage?: number | 'level';
+  defensiveCategory?: MoveCategory;
+  drain?: [number, number]; // fraction
+  forceSTAB?: boolean;
+  forceSwitch?: boolean;
+  givesHealth?: boolean;
+  hasCustomRecoil?: boolean;
+  hasPriority?: boolean;
+  hasSecondaryEffect?: boolean;
+  heal?: [number, number]; // fraction
+  ignoreAbility?: boolean;
+  ignoreAccuracy?: boolean;
+  ignoreDefensive?: boolean;
+  ignoreEvasion?: boolean;
+  ignoreImmunity?: boolean | TypeT;
+  ignoreNegativeOffensive?: boolean;
+  ignoreOffensive?: boolean;
+  ignorePositiveDefensive?: boolean;
+  ignorePositiveEvasion?: boolean;
+  ignoresBurn?: boolean;
+  ignoresDefenseBoosts?: boolean;
+  infiltrates?: boolean;
+  isFutureMove?: boolean;
+  isSpread?: boolean | 'allAdjacent';
+  isZ?: ItemT;
+  multiaccuracy?: boolean;
+  multihit?: number | [number, number]; // range
+  noCopy?: boolean;
+  noDamageVariance?: boolean;
+  noFaint?: boolean;
+  noMetronome?: MoveT[];
+  noPPBoosts?: boolean;
+  noSketch?: boolean;
+  nonGhostTarget?: string;
+  ohko?: true | 'Ice'; // TODO: better type here?
+  percentHealed?: number;
+  pressureTarget?: string;
+  priority?: number;
+  pseudoWeather?: string:
+  recoil?: [number, number]; // fraction
+  secondaries?: SecondaryEffect[];
+  self?: SelfEffect;
+  selfBoost?: {boosts?: Partial<BoostsTable>};
+  selfSwitch?: ID | boolean;
+  selfdestruct?: 'always' | 'ifHit';
+  sideCondition?: ID;
+  sleepUsable?: boolean;
+  slotCondition?: string;
```

Comment:
appears to be PS implementation detail

Reply:
It's relevant for "sets a slot condition"...

Reply:
It's useful for others to know "this move starts an effect on a specific pokemon slot that lasts through switches" (like Wish or Future Sight).

Reply:
`Wish` has `slotCondition: 'Wish'`, `Healing Wish` has `slotCondition: 'healingwish'`, and it's unused elsewhere. I don't know what this information conveys other than the name of the move it is attached to. You could just as easily replace it with `hasSlotCondition: boolean`, and use the move name

Reply:
It conveys that you can use `Dex.getEffect('healingwish')` to get information about the slot condition effect. It's the same format as other effect inducers, some of which induce an effect not named after the move inducing it (more than one thing induces Attract, lots of things induce flinch, multiple things induce weather, etc).

Reply:
It's also specifically a hitResult, marking that it can be on `move`, `move.secondary`, or `move.self`.

Reply:
volatileCondition/slotConditions/sideConditions/fieldConditions (ie/ weather/terrain) are all basically the same concept but with different scopes as to what they apply to, and different names. They (or some other field) are definitely required for certain things like `flinch` or `confusion`, and I would argue they are useful for things independent than just the move/ability that caused them (weathers and terrains can be triggered via abilities or moves). 

I think chaos is suggesting that the information is redundant in the more common case where the move name/ID (we arent consistent!) === the condition name. It seems like in those cases we could get by with `hasVolatileStatus: true | 'flinch' | 'confusion' | ...` and simply know that if the field is `true` we can just do `Dex.getEffect(move.id)` instead.

I don't disagree with this, but this would be a large change to the PS codebase and wouldn't add much value, because `'hasSlotCondition': true` is pretty much the exact same size as `'slotCondition': 'wish'` (or at least not meaningfully different), so I don't really see much argument for complicating the type and changing the status quo. Migrating PS this new native API is already going to be a herculean effort, I think making some affordances which dont impact performance or readability much should be accepted. Its also entirely possible that we can followup to try and find a better pattern for these conditions later - this API strawman is not going to live on unchanged forever after.